### PR TITLE
feat: use observed equations for guesses of observed variables

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: test/initializationsystem.jl [hunk 1]
```diff
@@ -836,3 +836,11 @@ end
     integ = init(prob, Rosenbrock23())
     @test integ[y] ≈ -0.5
 end
+
+@testset "Use observed equations for guesses of observed variables" begin
+    @variables x(t) y(t) [state_priority = 100]
+    @mtkbuild sys = ODESystem(
+        [D(x) ~ x + t, y ~ 2x + 1], t; initialization_eqs = [x^3 + y^3 ~ 1])
+    isys = ModelingToolkit.generate_initializesystem(sys)
+    @test isequal(defaults(isys)[y], 2x + 1)
```
Comment (ChrisRackauckas) [inline test/initializationsystem.jl:845]: but in this case `x` doesn't have a guess, so how would it work?

Comment (AayushSabharwal) [reply test/initializationsystem.jl:845]: The system wasn't meant to be solved, just to ensure that `y` has a guess using the observed equation.

Comment (ChrisRackauckas) [reply test/initializationsystem.jl:845]: I see but it's still not solvable?

Comment (AayushSabharwal) [reply test/initializationsystem.jl:845]: I can add a guess for `x` but I don't see how that helps the test 😅 the initializationsystem won't be solved during the test anyway

**Pull request merged**